crates/wdk-build/src/lib.rs

```diff
@@ -623,6 +629,134 @@ impl Config {
+                vec!["windows.h"]
+            }
+        }
+        .into_iter()
+        .map(std::string::ToString::to_string)
+    }
+
+    pub fn wdf_headers(&self) -> impl Iterator<Item = String> {
+        ["wdf.h"].into_iter().map(std::string::ToString::to_string)
+    }
+
+    #[cfg(feature = "hid")]
+    pub fn hid_headers(&self) -> impl Iterator<Item = String> {
+        // HID Headers list from https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/_hid/
+        {
+            let mut hid_headers = vec!["hidclass.h", "hidsdi.h", "hidpi.h", "vhf.h"];
+            if let DriverConfig::Wdm | DriverConfig::Kmdf(_) = self.driver_config {
+                hid_headers.extend(["hidpddi.h", "hidport.h", "kbdmou.h", "ntdd8042.h"]);
+            }
+            if let DriverConfig::Kmdf(_) = self.driver_config {
+                hid_headers.extend(["HidSpiCx/1.0/hidspicx.h"]);
+            }
+            hid_headers
+        }
+        .into_iter()
+        .map(std::string::ToString::to_string)
+    }
+
+    pub fn bindgen_base_header_contents(&self) -> String {
+        let mut header_contents = self.base_headers().fold(String::new(), |mut acc, header| {
+            acc.push_str(r#"#include ""#);
+            acc.push_str(&header);
+            acc.push_str("\"\n");
+            acc
+        });
+
+        if let DriverConfig::Wdm | DriverConfig::Kmdf(_) = self.driver_config {
+            // TODO: Why is there no definition for this struct? Maybe blocklist this struct
+            // in bindgen.
+            header_contents.push_str(
+                r"
+typedef union _KGDTENTRY64
+{
+  struct
+  {
+    unsigned short LimitLow;
+    unsigned short BaseLow;
+    union
+    {
+      struct
+      {
+        unsigned char BaseMiddle;
+        unsigned char Flags1;
+        unsigned char Flags2;
+        unsigned char BaseHigh;
+      } Bytes;
+      struct
+      {
+        unsigned long BaseMiddle : 8;
+        unsigned long Type : 5;
+        unsigned long Dpl : 2;
+        unsigned long Present : 1;
+        unsigned long LimitHigh : 4;
+        unsigned long System : 1;
+        unsigned long LongMode : 1;
+        unsigned long DefaultBig : 1;
+        unsigned long Granularity : 1;
+        unsigned long BaseHigh : 8;
+      } Bits;
+    };
+    unsigned long BaseUpper;
+    unsigned long MustBeZero;
+  };
+  unsigned __int64 Alignment;
+} KGDTENTRY64, *PKGDTENTRY64;
+
+typedef union _KIDTENTRY64
+{
+  struct
+  {
+    unsigned short OffsetLow;
+    unsigned short Selector;
+    unsigned short IstIndex : 3;
+    unsigned short Reserved0 : 5;
+    unsigned short Type : 5;
+    unsigned short Dpl : 2;
+    unsigned short Present : 1;
+    unsigned short OffsetMiddle;
+    unsigned long OffsetHigh;
+    unsigned long Reserved1;
+  };
+  unsigned __int64 Alignment;
+} KIDTENTRY64, *PKIDTENTRY64;
+",
+            );
+        }
+
+        header_contents
+    }
+
+    pub fn bindgen_wdf_header_contents(&self) -> Option<String> {
```

Comment:
this should probably be a common `bindgen_header_contents` function that takes in some enum value for each of the header groups